satpy/dependency_tree.py

```diff
@@ -107,7 +107,8 @@ def add_child(self, parent, child):
         #               multiple times if more than one Node depends on them
         #               but they should all map to the same Node object.
         if self.contains(child.name):
-            assert self._all_nodes[child.name] is child
+            if self._all_nodes[child.name] is not child:
+                raise AssertionError
```

Comment:
What is the benefit of this change?

Reply:
Python has an option to compile the optimized bytecode and create the respective `.pyo` files by using the options `-O` and `-OO`.

During this, these basic optimizations are done:
- All the assert statements are removed
- All docstrings are removed (when `-OO` is selected)
- Value of the __debug__ built-in variable is set to False

Now, if someone never intends to run the optimized bytecode, they can ignore this issue. But, it is still recommended not to have it in non-test files.
A better way for internal self-checks is to explicitly check and raise respective Error using an if statement.

Reply:
Interesting, thanks, I wasn't aware of this.
In this case, the Assertion error should be replaces with a RuntimeError I suppose...

Reply:
I'm a little unsure on this change. I knew about the assertions being removed and figured this was a reasonable "optimization" in the Satpy code as this line should never be triggered...but then why have it I suppose. I guess I agree with @mraspaud that this should be something like a `RuntimeError`.

Reply:
So should I replace all occurrences with `RuntimeError`?

Reply:
Yes, that would be great.
Are all these changes you make configurable in deepsource?